webapp/views.py

```diff
@@ -1037,7 +1042,32 @@ def tutorials_index():
         page = flask.request.args.get("page", default=1, type=int)
         topic = flask.request.args.get("topic", default=None, type=str)
         sort = flask.request.args.get("sort", default=None, type=str)
+        query = flask.request.args.get("q", default=None, type=str)
         posts_per_page = 15
+
+        """
+        Get search results from Google Custom Search
+        """
+
+        # The webteam's default custom search ID
+        search_engine_id = "009048213575199080868:i3zoqdwqk8o"
+
+        # API key should always be provided as an environment variable
+        search_api_key = os.getenv("SEARCH_API_KEY")
+
+        if not search_api_key:
+            raise NoAPIKeyError("Unable to search: No API key provided")
```

Comment:
It's actually a shame you have to do this again here. It would be nice if the code inside the search module was encapsulated such that it contained all this logic but allowed you to get results for your own view.

I'll file an issue. But this is fine for now.

Reply:
Filed https://github.com/canonical-web-and-design/canonicalwebteam.search/issues/5